cmd/server/pactasrv/analysis.go

```diff
@@ -341,3 +283,88 @@ func (s *Server) RunAnalysis(ctx context.Context, request api.RunAnalysisRequest
 
 	return api.RunAnalysis200JSONResponse{AnalysisId: string(result)}, nil
 }
+
+func (s *Server) analysisDoAuthzAndAuditLog(ctx context.Context, analysisID pacta.AnalysisID, action pacta.AuditLogAction) error {
```

Comment:
optional: All these helpers could return the entity they've loaded, saves you a DB call on the GET path.

Reply:
I thought about that, and ultimately decided against it just for simplicity - If we end up seeing user-visible performance as a result I'm totally up for it, but I love the simplicity of the same method signature without a few `_, _, _, err := ...` floating around (which would be the case to obviate things like membership lookups)